src/components/promptOAuth/PromptOAuth.js

```diff
@@ -0,0 +1,95 @@
+import React, {Fragment} from 'react'
+import PropTypes from 'prop-types'
+import {Modal} from "@instructure/ui-modal";
+import {Heading} from "@instructure/ui-heading";
+import {View} from "@instructure/ui-view";
+import {Button} from "@instructure/ui-buttons";
+import {InstUISettingsProvider} from "@instructure/emotion";
+
+/**
+ * Displays a modal asking the user to grant access when requested to.
+ */
+export class PromptOAuth extends React.Component {
+
+    static propTypes = {
+        /**
+         * The token used for requests against the proxy.
+         */
+        accessToken: PropTypes.string,
+        /**
+         * Function called after user has granted the application access to their account.
+         */
+        tokenGranted: PropTypes.func.isRequired,
+        /**
+         * Should we prompt the user to grant access.
+         */
+        needsGrant: PropTypes.bool.isRequired,
+        /**
+         * The proxy server address.
+         */
+        server: PropTypes.string.isRequired,
+        /**
+         * The name of the tool requesting access.
+         */
+        toolName: PropTypes.string
+
+    }
+
+    static defaultProps = {
+        accessToken: "",
+    }
+
+    componentDidMount() {
+        window.addEventListener("message", (event) => {
+            if (event.data === 'token') {
+                this.props.tokenGranted()
+            }
+        }, false)
+    }
+
+    render() {
+
+        const {server, accessToken, needsGrant, tokenGranted, toolName} = this.props
+        const title = toolName ? `Access Required: ${toolName}` : `Access Required`
+        return <>
+            <InstUISettingsProvider
+                theme={{
+                    themeOverrides: {
+                        canvas: {
+                            componentOverrides: {
+                                Mask: {
+                                    // This is to grey out the background behind the modal more so that it's clear
+                                    // what content is waiting on the OAuth grant. For example in the Module Titles
+                                    // there may be other content in the page that doesn't require the granting of 
+                                    // permission
+                                    background: 'rgba( 200, 200, 200, 0.5 )'
```

Comment:
css or we do stuff inline too these days if it's one-off/small?

Reply:
Sorry, what are you suggesting here? We need to override the theme variables as documented: https://instructure.design/#using-theme-overrides

Reply:
ah ok